contrib/hyperscan/matching/input_matchers/source/matcher.h

```diff
@@ -14,26 +13,42 @@ namespace InputMatchers {
 namespace Hyperscan {
 
 struct ScratchThreadLocal : public ThreadLocal::ThreadLocalObject {
-  explicit ScratchThreadLocal(const hs_database_t* database);
-  ~ScratchThreadLocal() override { hs_free_scratch(scratch_); }
+  ScratchThreadLocal(const hs_database_t* database, const hs_database_t* som_database);
+  ~ScratchThreadLocal() override;
+
   hs_scratch_t* scratch_{};
 };
 
-class Matcher : public Envoy::Matcher::InputMatcher {
+struct Bound {
+  Bound(uint64_t start, uint64_t end);
+
+  bool operator<(const Bound& other) const;
+
+  uint64_t start_;
```

Comment:
I assume end_ is exclusive, like STL end() ?

If that's the case can you call these begin_ and end_ ?

In either case can you comment?

Reply:
It does not matter, can be called whatever you want. For me, it is just a struct holds two variables which indicates the start and end number. It will be fine to change to `begin_`.